### PR TITLE
Mixed-type feeds

### PR DESCRIPTION
**Pull request by jimrandomh**
Recent Discussion: Now has infinite scroll Has recently-commented-on tags, following grouping and bumping rules similar to posts. Has edits to tags that add more than 100 characters.

Tag/Wiki Activity Feed: There is now a tag activity feed at `/tagFeed`. It contains all edits to and comments on tags/wiki pages, and all tag creations, sorted by recency.

Tag history: New page at `/tag/tagname/history`. This is an infinitely-scrolling list of events for a specific tag/wiki page, including creation, revisions, discussion threads (grouped by thread, sorted by root comment date), and tag applications to posts. This is planned to replace the existing revision-history page when it's debugged.

Known issue that I'm still working on: When you reply to things in the Recent Discussion feed, your comment does not show up until you refresh.

### CODE REVIEW

File: packages/lesswrong/components/tagging/ChangeMetricsDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { registerComponent } from '../../lib/vulcan-lib';
+import { ChangeMetrics } from '../../lib/collections/revisions/collection';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  charsAdded: {
+    color: "#008800",
```
[the remark this answers is not in the document]
Comment (jimrandomh) [reply packages/lesswrong/components/tagging/ChangeMetricsDisplay.tsx:7]: Factored out from elsewhere, yes; this should be Ray's color choice.

File: packages/lesswrong/components/recentDiscussion/RecentDiscussionTag.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+              startThreadTruncated={true}
+              expandAllThreads={initialExpandAllThreads || expandAllThreads}
+              scrollOnExpand
+              nestingLevel={1}
+              lastCommentId={lastCommentId}
+              comment={comment.item}
+              markAsRead={markAsRead}
+              highlightDate={lastVisitedAt}
+              //eslint-disable-next-line react/no-children-prop
+              children={comment.children}
```
[the remark this answers is not in the document]
Comment (jimrandomh) [reply packages/lesswrong/components/recentDiscussion/RecentDiscussionTag.tsx:133]: Yep, the prop should not be renamed `children`. This is a preexisting issue, but now seems like a good time to do the rename.

File: packages/lesswrong/components/common/withOnPageScroll.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  React.useEffect(() => {
+    if (Meteor.isClient) {
+      document.addEventListener('scroll', onScrollFn)
+      
+      return function cleanup() {
+        document.removeEventListener('scroll', onScrollFn);
+      };
+    }
+  });
+}
```
[the remark this answers is not in the document]
Comment (jimrandomh) [reply packages/lesswrong/components/common/withOnPageScroll.tsx:14]: Fixed

**Comment by jimrandomh** [comment]
Modified to now us Apollo's `fetchMore` rather than a more-complicated system I had before with mounting components and callbacks. This worked better than I expected (I lose a bet on this with Oli); the only issue I encountered is that the `loading` state doesn't reflect whether a `fetchMore` is in flight, which requires a little bit of machination to keep duplicate requests from going out, but it wasn't too bad.

Keeping everything in Apollo means that comment-editing works correctly without explicit refetches. Anything involving creating a new comment causes Recent Discussion to refetch entirely from scratch..

File: packages/lesswrong/components/common/MixedTypeFeed.tsx [hunk 4]
```diff
@@ -74,68 +95,123 @@ const MixedTypeFeed = (args: {
...
-  const query = getQuery({resolverName, resolverArgs, fragmentArgs, sortKeyType, renderers});
   
-  const {data: firstPageData, error: firstPageError} = useQuery(query, {
+  // Reference to a bottom-marker used for checking scroll position.
+  const bottomRef = useRef<HTMLDivElement|null>(null);
+  
+  // Whether there is a pending query. This is indirect (inside an object)
+  // because it's accessed from inside callbacks, where the timing of state
+  // updates would be a problem.
+  const [queryIsPending] = useState({isPending: false});
```
Comment (Discordius) [inline packages/lesswrong/components/common/MixedTypeFeed.tsx:110]: Change this to use `useRef`

**Review by Discordius — APPROVED** [review]

**Pull request merged**